Mathlib/Topology/Homotopy/Product.lean

```diff
@@ -177,36 +182,40 @@ This is `trans_prod_eq_prod_trans` descended to the quotient. -/
 theorem comp_prod_eq_prod_comp : prod q₁ q₂ ⬝ prod r₁ r₂ = prod (q₁ ⬝ r₁) (q₂ ⬝ r₂) := by
   induction q₁, q₂ using Quotient.inductionOn₂
   induction r₁, r₂ using Quotient.inductionOn₂
-  simp only [prod_lift, ← Path.Homotopic.comp_lift, Path.trans_prod_eq_prod_trans]
+  simp only [Quotient.mk''_eq_mk]
```

Comment:
This step (and a few others similar below) can be avoided by using a correct induction principle (such as `Path.Homotopic.Quotient.ind`) instead of induction principles from `_root_.Quotient`. You already added `ind` as an alias for `Quot.ind`, so maybe having a two-var induction principle would be nice as well? This slightly increases the amount of duplication though.

Reply:
Indeed, it's worthwhile. Done.